src/components/policies.tsx

```diff
@@ -0,0 +1,62 @@
+import { unitDetails, Policy } from "./interface";
+import {
+  COUNTABLE_HOUSEHOLD_STATUS,
+  HOUSEHOLD_TYPES,
+  STATUS_CODES,
+  NOT_HOME_STATUS_CODES,
+  HOUSEHOLD_LANGUAGES
+} from "./util";
+
+export class RacePolicy implements Policy {
+  isCountable(unit: unitDetails): boolean {
+    return (
+      COUNTABLE_HOUSEHOLD_STATUS.includes(unit.status as string) &&
+      unit.type !== HOUSEHOLD_TYPES.MALAY
+    );
+  }
+  isCompleted(unit: unitDetails): boolean {
+    return (
+      unit.status === STATUS_CODES.DONE ||
+      (unit.status === STATUS_CODES.NOT_HOME &&
+        (unit.nhcount === NOT_HOME_STATUS_CODES.SECOND_TRY ||
+          unit.nhcount === NOT_HOME_STATUS_CODES.THIRD_TRY))
+    );
+  }
+}
+
+export class LanguagePolicy implements Policy {
+  maxTries: number;
+  homeLanguage: string;
+  constructor(
+    maxtries = parseInt(NOT_HOME_STATUS_CODES.SECOND_TRY),
+    homelanguage = HOUSEHOLD_LANGUAGES.ENGLISH.CODE
+  ) {
+    this.maxTries = maxtries;
+    this.homeLanguage = homelanguage;
+  }
+  isHomeLanguage(unit: unitDetails): boolean {
+    const languageValue = unit.languages.toUpperCase().trim();
```

Comment:
There are quite a no. of string manipulations(2x toUppercase & trim) in this function.
If capitalised language code is your go-to format, I think it would be more efficient to set them in source rather than converting it in the frontend. @bumblebeelabs 
There will be another PR. I'll merge this PR first.